mailet/standard/src/test/java/org/apache/james/transport/matchers/AtMostTest.java

```diff
@@ -56,103 +57,208 @@ private Mail createMail() throws MessagingException {
-
-        assertThat(actual).containsOnly(RECIPIENT1);
+    @Nested
+    class WrongConditionConfigurationTest {
+        @Test
+        void shouldThrowWhenMatchersConfigWithOutConditionValue() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition("randomName:")
+                .build()))
+                .isInstanceOf(MessagingException.class);
+        }
+
+        @Test
+        void shouldThrowWhenMatchersConfigWithConditionValueAsWord() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition("value")
+                .build()))
+                .isInstanceOf(MessagingException.class);
+        }
+
+        @Test
+        void shouldThrowWhenMatchersConfigWithNegativeConditionValue() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition("-87")
+                .build()))
+                .isInstanceOf(MessagingException.class);
+        }
+
+        @Test
+        void shouldThrowWhenMatchersConfigWithoutConditionValue() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition("randomName:")
+                .build()))
+                .isInstanceOf(MessagingException.class);
+        }
+
+        @Test
+        void shouldThrowWhenMatchersConfigWithoutConditionName() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition(":3")
+                .build()))
+                .isInstanceOf(IllegalArgumentException.class);
+        }
+
+        @Test
+        void shouldThrowWhenMatchersConfigNameAsSpace() {
+            assertThatThrownBy(() -> new AtMost().init(FakeMatcherConfig.builder()
+                .matcherName("NoValueMatcher")
+                .condition("  :  ")
+                .build()))
+                .isInstanceOf(MessagingException.class);
+        }
     }
 
-    @Test
-    void shouldMatchWhenNoRetries() throws MessagingException {
-        Mail mail = createMail();
-        mail.setAttribute(new Attribute(AT_MOST_TRIES, AttributeValue.of(0)));
+    @Nested
+    class MultiplesMatcherConfigurationTest {
+        private AtMost multipleMatchers;
 
-        Collection<MailAddress> actual = matcher.match(mail);
+        @BeforeEach
+        void setup() throws MessagingException {
+            this.multipleMatchers = new AtMost();
+            multipleMatchers.init(
+                FakeMatcherConfig.builder()
+                    .matcherName("MultipleMatchers")
+                    .condition("randomName:3")
+                    .condition("randomName:5")
```

Comment:
what are you trying to achieve here? If you want two matchers, build two matchers. If you look in the builder, you will see there is only one condition. So you probably end up here with one matcher with one condition to `randomName:5`

Reply:
Please specify only one condition in the FakeMatcherConfig.Builder